File: plugins/minimum.py
import re
from plugin import Plugin


class MinimumPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'最小\s+(.+)', line)

        if not match:
            return None

        values = match.group(1)

        return f"print(min({values}))"

File: web/plugins/minimum.js
import { WebPlugin } from "../engine/plugin.js";

export class MinimumPlugin extends WebPlugin {
    constructor() {
        super("最小");
    }

    async execute(line, context) {
        const expression = line
            .replace(/^最小\s*/, "");

        const values = context.parser
            .splitArguments(expression)
            .map(value =>
                Number(
                    context.parser.parseValue(
                        value,
                        context.runtime
                    )
                )
            );

        context.runtime.write(
            Math.min(...values)
        );
    }
}